students/JonSerpas/lesson09/assignment/src/jpgdiscover.py

```diff
@@ -0,0 +1,24 @@
+import os
+from pprint import pprint
+
+data = {}
+
+
+def find_files(folder, file_extension):
+    for item in os.listdir(folder):
+        item = "/".join([folder, item])
+        if os.path.isdir(item):
+            data[item] = []
+            find_files(item, file_extension)  # RECURSION TIME
+            if len(data[item]) == 0:
+                del data[item]
+        else:
+            if item[-1*len(file_extension):] == file_extension:
+                # print(item)
+                basename = os.path.basename(item)
+                dirname = os.path.dirname(item).strip("/")
+                data[dirname].append(basename) #key error here, dunno why?
```

Comment:
If you're seeing this key error, it's seems that the dirname doesn't exist in data. It may be useful to catch a key error and print out the key for that case so you can see which dirname is at fault.